testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/userdata/ContactsActivity.kt

```diff
@@ -0,0 +1,82 @@
+package com.rakuten.tech.mobile.testapp.ui.userdata
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import android.view.MenuItem
+import androidx.appcompat.widget.Toolbar
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.rakuten.tech.mobile.miniapp.testapp.R
+import com.rakuten.tech.mobile.testapp.AppScreen
+import com.rakuten.tech.mobile.testapp.helper.clearWhiteSpaces
+import com.rakuten.tech.mobile.testapp.ui.base.BaseActivity
+import com.rakuten.tech.mobile.testapp.ui.settings.SettingsMenuActivity
+import kotlinx.android.synthetic.main.contacts_activity.*
+import java.util.UUID
+
+class ContactsActivity : BaseActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.contacts_activity)
+        initializeActionBar()
+        createContactList()
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            android.R.id.home -> {
+                navigateToPreviousScreen()
+                return true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    private fun initializeActionBar() {
+        val toolBar = findViewById<Toolbar>(R.id.toolbar)
+        setSupportActionBar(toolBar)
+        showBackIcon()
+    }
+
+    private fun createContactList() {
```

Comment:
Also, the contact list needs to be editable - the user should be able to remove contacts or add a new contact. On iOS Demo App when the user adds a new contact, an input dialog is shown where they can enter the ID.

This is because the Contact IDs are arbitrarily generated values within the Host App, so they might not be just UUIDs - they could be any format. So we need to be able to QA with a variety of different formats and contact list lengths.

Reply:
> There is a use case to share contact list so we need to have a fixed contact list. We can save the random list in storage after first time randomly generated.

I have added a function get the current contact list from the adapter, so we can use it for caching.

```
interface ContactAdapterPresenter {
    ...
    fun provideContactEntries(): ArrayList<String>
}

```

Do we need to cache the contact list in this PR?

Reply:
> Also, the contact list needs to be editable - the user should be able to remove contacts or add a new contact. On iOS Demo App when the user adds a new contact, an input dialog is shown where they can enter the ID.

I have update this part with the screenshots in the PR's description how it looks like now.

Reply:
I think, we should cache the contact list because we will able to the saved contacts while coming back to this page again.

Reply:
Yes, we need to save the contact list in storage.